api/pkg/apps/iam/auth.go

```diff
@@ -11,6 +12,18 @@ func (s *Server) WithAuth() func(handler http.Handler) http.Handler {
 	return func(handler http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 
+			ctx := req.Context()
+
+			if s.environment == "Development" {
+				authUserSubject := req.Header.Get("X-Authenticated-User-Subject")
+				if len(authUserSubject) != 0 {
+					ctx = context.WithValue(ctx, "Subject", authUserSubject)
+					req = req.WithContext(ctx)
+					handler.ServeHTTP(w, req)
+					return
+				}
+			}
+
```

Comment:
Can you extract this to `pkg/testing/auth.go`  ? Sonarcloud is complaining that this is duplicated